imap/http_caldav.c

```diff
@@ -3652,9 +3652,13 @@ static int validate_dtend_duration(icalcomponent *comp, struct error_t *error)
             return HTTP_FORBIDDEN;
         }
         if (icaltime_compare(dtend, dtstart) < 1) {
-            error->desc = "DTEND must occur after DTSTART";
-            error->precond = CALDAV_VALID_DATA;
-            return HTTP_FORBIDDEN;
+            if (icaltime_compare(dtend, dtstart) < 0) {
```

Comment:
How about storing the result of icaltime_compare() in a variable once, and the use the variable twice?

Reply:
What means only used by Fastmail?  Everbody using the public code does have the extra calculation.

Reply:
This is a pull request, NOT code in master